sql/insert.go

```diff
@@ -93,24 +115,27 @@ func (p *planner) Insert(n *parser.Insert) (planNode, error) {
 		return nil, err
 	}
 	if err := p.db.Run(&b); err != nil {
+		if tErr, ok := err.(*proto.ConditionFailedError); ok {
+			return nil, fmt.Errorf("duplicate key value %q violates unique constraint %s", tErr.ActualValue.Bytes, "TODO(tamird)")
+		}
```

Comment:
The error returned by `p.db.Run(&b)` is the first error encountered in the batch. I think you can loop over `b.Results` to find the first one that returned an error and match it up with the index being modified. Irritating, but likely worthwhile. 

You definitely need to be running the batch within a transaction. Could have sworn that was being done already.